engineering_tools/carma_script_extensions/cloud-sim

```diff
@@ -217,112 +209,113 @@ cloud-sim__start() {
+            ((i++))
 
-    done
+            # Wait SLEEP_SEC seconds before checking again
+            sleep ${SLEEP_SEC}
+            ELAPSED_TIME=$((SLEEP_SEC * i))
 
-    if [ ${ELAPSED_TIME} -gt ${LAUNCH_TIMEOUT} ]; then
-        echo "Stack launch timedout. Inspect AWS console for details. Remember to delete the stack after debugging!"
-        exit 1
-    fi
+        done
 
+        if [ ${ELAPSED_TIME} -gt ${LAUNCH_TIMEOUT} ]; then
+            echo "Stack launch timedout. Inspect AWS console for details. Remember to delete the stack after debugging!"
+            exit 1
+        fi
 
-    local DEV_MACHINE_ID=$(aws cloudformation describe-stack-resources --stack-name ${STACK_NAME} --output text --logical-resource-id DevMachine --query StackResources[].PhysicalResourceId)
 
-    local PUBLIC_DNS=$(aws ec2 describe-instances --instance-ids ${DEV_MACHINE_ID} --output text --query Reservations[].Instances[].PublicDnsName)
+        local DEV_MACHINE_ID=$(aws cloudformation describe-stack-resources --stack-name ${STACK_NAME} --output text --logical-resource-id DevMachine --query StackResources[].PhysicalResourceId)
 
-    local CLOUD9_ADDRESS_BASE="https://us-east-1.console.aws.amazon.com/cloud9/ide"
-    # 1. Extract cloud9 name from stack details
-    local CLOUD9_ENVS=$(aws cloud9 list-environments --output text | sed "s|ENVIRONMENTIDS\s||")
-    # 2. Determine cloud9 environment id
-    local CLOUD9_NAME="RobotWorkshop-${STACK_NAME}"
-    local CLOUD9_ID=$(aws cloud9 describe-environments --output text --environment-ids $(echo $CLOUD9_ENVS) --query "environments[?name=='${CLOUD9_NAME}'].id")
-    # 3. Update URL
-    local CLOUD9_ADDRESS="${CLOUD9_ADDRESS_BASE}/${CLOUD9_ID}"
+        local PUBLIC_DNS=$(aws ec2 describe-instances --instance-ids ${DEV_MACHINE_ID} --output text --query Reservations[].Instances[].PublicDnsName)
 
-    # Save the stack name for cleanup
-    echo "STACK_NAME=${STACK_NAME}" > ${CARMA_CLOUD_SIM_STACK_PATH}
+        local CLOUD9_ADDRESS_BASE="https://us-east-1.console.aws.amazon.com/cloud9/ide"
+        # 1. Extract cloud9 name from stack details
+        local CLOUD9_ENVS=$(aws cloud9 list-environments --output text | sed "s|ENVIRONMENTIDS\s||")
+        # 2. Determine cloud9 environment id
+        local CLOUD9_NAME="RobotWorkshop-${STACK_NAME}"
+        local CLOUD9_ID=$(aws cloud9 describe-environments --output text --environment-ids $(echo $CLOUD9_ENVS) --query "environments[?name=='${CLOUD9_NAME}'].id")
+        # 3. Update URL
+        local CLOUD9_ADDRESS="${CLOUD9_ADDRESS_BASE}/${CLOUD9_ID}"
+
+        # Save the stack name for cleanup
+        echo "STACK_NAME=${STACK_NAME}" > ${CARMA_CLOUD_SIM_STACK_PATH}
+        
+        # Print for user
+        echo "##############################"
+        echo ""
+        echo "AWS Stack ${STACK_NAME} Launch Suceeded"
+        echo ""
+        echo "EC2 Instance Address: ${PUBLIC_DNS}"
+        echo ""
+        echo "Cloud9 IDE Address: ${CLOUD9_ADDRESS}"
```

Comment:
Out of curiosity, did you confirm that the Cloud9 config still works? Not sure it's super high priority, but it is something that's nice for development.

Reply:
I just tested this and it gave me an error trying to access Cloud9
"1 validation error detected: Value 'ide' at 'environmentId' failed to satisfy constraint: Member must satisfy regular expression pattern: ^[a-zA-Z0-9]{8,32}$"

Reply:
Issue logged here https://github.com/usdot-fhwa-stol/carma-platform/issues/2169